app/controllers/floors_controller.rb

```diff
@@ -141,6 +141,7 @@ def process_search(search_params)
     results << Location.search_for(search_params)
     results << Tag.search_for(search_params).has_location
     results << Trait.search_for(search_params).has_location
+    results << Icon.search_for(search_params).select{ |i| !i.location_ids.nil? }
```

Comment:
use the scope you created instead

Reply:
So the scope wasnt working because icons dont have the location_id column. So it cant do the database search. Should I update the schema to include location_ids? If so, can a database table map to an array of ids? I was kind of concerned about that.